### PR TITLE
DietPi-Software | VSCodium

### LINKED ISSUE #4267: DietPi-Software | VSCodium
#### Formal software information
- Software title | 
VSCodium
- Short description | 
FLOSS version of MS VScode
- Official URL (if available) | 
https://vscodium.com/
- GitHub URL (if available) | 
https://github.com/VSCodium/vscodium
- (Official) install documentation | 
https://gitlab.com/paulcarroty/vscodium-deb-rpm-repo#option-1-recommended

#### Are there similar/alternative software titles available with DietPi-Software?

- Vim, Jed

#### What makes your requested software better than the above solutions, if available?

- GUI software, many features

#### How can DietPi make the installation easier or compatible, than following the install instructions or do APT installation, if available?

- Have to add an APT repository.

#### Can you provide the installation steps that you would suggest DietPi-Software to do?

1. Add APT repository
2. `G_AGI codium`

#### Are you willing to help maintaining the software installation, e.g. in case of needed setup changes due to updates etc.? This is not needed, but could speed up our decision to implement it, as man power is always a topic :wink:.

- Probably

### PR DESCRIPTION
**Pull request by nonnorm**
**Status**: Ready

**Reference**: https://github.com/MichaIng/DietPi/issues/4267

**Commit list/description**:

- DietPi-Software | Add VSCodium install option
- DietPi-Software | Make VSCodium require desktop
- DietPi-Software | Also make VSCodium require git

### CODE REVIEW

**Comment by nonnorm** [comment]
Well, that was simple.

File: dietpi/dietpi-software [hunk 1]
```diff
@@ -5496,6 +5504,23 @@ Package: wireguard wireguard-dkms wireguard-tools\nPin: release n=bullseye\nPin-
...
+		software_id=189 # VSCodium
+		if (( ${aSOFTWARE_INSTALL_STATE[$software_id]} == 1 )); then
+
+			Banner_Installing
+			
+			# Install APT repo key
+			G_EXEC eval "curl -sSfL 'https://gitlab.com/paulcarroty/vscodium-deb-rpm-repo/raw/master/pub.gpg' | gpg --dearmor -o /etc/apt/trusted.gpg.d/dietpi-vscodium.gpg --yes"
+
+			# Install APT repo list
+			G_EXEC eval "echo 'deb https://paulcarroty.gitlab.io/vscodium-deb-rpm-repo/debs vscodium main' > /etc/apt/sources.list.d/vscodium.list"
```
Comment (MichaIng) [inline dietpi/dietpi-software]: Let's use `dietpi-vscodium.list`. I know we skip the prefix for most lists somehow, but I aim to use it consequently to make identifying easily which files are coming from us and which were created manually or by other installers/packages.

File: dietpi/dietpi-software [hunk 2]
```diff
@@ -15362,6 +15387,17 @@ _EOF_
 
 		fi
 
+		software_id=189 # VSCodium
+		if (( ${aSOFTWARE_INSTALL_STATE[$software_id]} == -1 )); then
+
+			Banner_Uninstalling
+
+			G_AGP codium
+			[[ -f '/etc/apt/sources.list.d/vscodium.list' ]] && G_EXEC_NOEXIT=1 G_EXEC rm /etc/apt/sources.list.d/vscodium.list
```
Comment (MichaIng) [inline dietpi/dietpi-software]: Rename as above.

**Comment by MichaIng** [comment]
Lol a pretty simple naked install, but that is all fine when the package-contained setup is good. Good to have this as an alternative to VSCode after #4083.

**Comment by Joulinar** [comment]
and the good thing, version provided via `apt` package is same as available on GitHub. 👍 

From install test
`Setting up codium (1.55.2-1618361277) ...`

on GitHub same version `1.55.2` is available https://github.com/VSCodium/vscodium/releases

**Comment by Joulinar** [comment]
I guess it's an issue of `VSCodium` but not sure. Could be as well me doing something wrong. At least I get an error while connecting to GitHub using the official `GitHub Pull Request` extension.

image

**Comment by MichaIng** [comment]
Please try to install/enable dbus:
```
apt install dbus
systemctl enable --now dbus
```
**EDIT: Ah no that cannot be the reason as it's installed and enabled automatically with all desktop installs.**

**Comment by nonnorm** [comment]
It looks like `gnome-keyring` is required, as after that I get no error.

**Comment by MichaIng** [comment]
When I try to sign into GitHub from VSCodium, in that status bar, it shows "Signing in to github.com..." forever, while it should show a login prompt, as far as I understand. I was thinking that it might be due to missing default browser, but making Firefox the default browser doesn't change anything 🤔.

Btw, we should add the application icon to the desktop?
_______
**EDIT: Tried to install `gnome-keyring`, but doesn't change something here. LXDE btw.**

**Comment by Joulinar** [comment]
You would need to click on `Signing in to github.com...` to open the field where you can enter GitHub token

**Comment by MichaIng** [comment]
Ahhh, I didn't recognised that at the top.
**EDIT: Worked for me without `gnome-keyring` as well.**

**Comment by Joulinar** [comment]
@MichaIng 

> EDIT: Worked for me without gnome-keyring as well.

You don't have the error message at the bottom?

**Comment by MichaIng** [comment]
Nope.

**Comment by Joulinar** [comment]
Hmm I need to do another test. Which extension you used?

**Comment by MichaIng** [comment]
The official GitHub PR and Issues extension. I'll retry to be sure.

**Comment by MichaIng** [comment]
Okay, it was still required on a fresh setup. I think that daemon is needed to create a new keyring with credentials, while accessing an existing one does not require it, or so.

I'm not sure whether there is an alternative to that package, other issues indicate it, but I couldn't find one when searching packages with -keyring ending at least.

Shall we add it by default or add a note to the docs instead? The package is small, but it's an additional daemon/process running.

**Comment by Joulinar** [comment]
depends on what people will use it for. I guess this is dedicated to GitHub? Or does it might be related to other plugins as well?

**Comment by nonnorm** [comment]
I think that it would probably be best to add it, as you will probably need it for signing in to Microsoft accounts as well.  However, we could probably configure the keychain automatically with the global software password.

**Comment by MichaIng** [comment]
I guess so. Cannot imagine that this extension uses its very own credentials/authentication method but that it's an API of codium that is use by other extensions as well.

**Comment by MichaIng** [comment]
Little research, `gnome-keyring` provides a backend daemon for the "Secrets API", described here: https://freedesktop.org/wiki/Specifications/secret-storage-spec/
Aside of the KDE wallet with GUI, there do not seem to be any relevant alternatives currently. Found https://github.com/grawity/secretsd, but it does not encrypt passwords, is hence more an experiment, I'd say.

Good to know about this in general, as other desktop applications might use that API as well to store or receive passwords, and also good to know that this is a central storage, hence you do not only trust the application like VSCodium here (which stores the master password) to safely store your GitHub token or other credentials but the Gnome keyring and API developers as well. Usually not relevant, but quite an important details for security enthusiasts 😉.

**Comment by nonnorm** [comment]
Hm, bash doesn't seem to be expanding the second `home/*` glob when uninstalling, and treating it literally.

**Comment by MichaIng** [comment]
Hmm , generally works fine here:
```
# bash -c 'rm -v /{root,home/*}/test'
rm: cannot remove '/root/test': No such file or directory
removed '/home/test/test'
```
I'll try it with VSCodium uninstall as well.

**Comment by nonnorm** [comment]
As a comparison:
```
# bash -c 'rm -v /{root,home/*}/test'
rm: cannot remove '/root/test': No such file or directory
rm: cannot remove '/home/*/test': No such file or directory
```
I'll try with a fresh VM.
**Edit**: It seems the glob only expands when the file/directory it's looking for exists, otherwise it prints it out literally.

**Comment by MichaIng** [comment]
> It seems the glob only expands when the file/directory it's looking for exists, otherwise it prints it out literally.

Ah yes of course. Taking the glob literal is basically part of the expansion. `*` matches the literal `*` as well. What we want is to hide "file not found" errors: https://github.com/MichaIng/DietPi/pull/4276/commits/02c8bb833a60889450083855b1cf47727223c733

**Review by MichaIng — APPROVED** [review]

**Review by Joulinar — APPROVED** [review]
going to approve the PR from inside `vscodium`

**Pull request merged**